fpga_interchange/arch.cc

```diff
@@ -83,13 +90,30 @@ void IdString::initialize_arch(const BaseCtx *ctx) {}
 
 static const ChipInfoPOD *get_chip_info(const RelPtr<ChipInfoPOD> *ptr) { return ptr->get(); }
 
+static std::string sha1_hash(const char *data, size_t size)
+{
+    boost::uuids::detail::sha1 hasher;
+    hasher.process_bytes(data, size);
+
+    unsigned int digest[5];
```

Comment:
nitpick, can you use `uint32_t` here as is convention in nextpnr when an explicit width is desired

Reply:
So it looks like `unsigned int` is the type that is used by `boost::uuids::detail::sha1`, see https://www.boost.org/doc/libs/1_70_0/boost/uuid/detail/sha1.hpp .  Do you still want me to change it to `uint32_t` in that case?

Reply:
I'm changing this to `boost::uuids::detail::sha1::digest_type`.

Reply:
`boost::uuids::detail::sha1::digest_type` seems reasonable

Reply:
Fixed